test/integration/cat/test_cat.py

```diff
@@ -49,3 +51,58 @@ def test_cat_xlog_file(tt_cmd, tmp_path):
     rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
     assert rc == 0
     assert re.search(r"replica_id: 1", output)
+
+
+@pytest.mark.parametrize("input, expected", [
+    pytest.param("abcdef", "failed to parse a timestamp: parsing time \"abcdef\" as"),
+    pytest.param("2024-11-14T14:02:36.abc", "parsing time \"2024-11-14T14:02:36.abc\" as"),
+])
+def test_cat_xlog_file_timestamp_incorrect(tt_cmd, tmp_path, input, expected):
+    # Copy the .xlog file to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "timestamp.xlog")
+    shutil.copy(test_app_path, tmp_path)
+
+    cmd = [tt_cmd, "cat", "timestamp.xlog", "--timestamp={0}".format(input)]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 1
+    assert re.search(r"{0}".format(expected), output)
+
+
+def test_cat_xlog_file_timestamp_clean(tt_cmd, tmp_path):
+    # Copy the .xlog file to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "timestamp.xlog")
+    shutil.copy(test_app_path, tmp_path)
+
+    cmd = [tt_cmd, "cat", "timestamp.xlog", "--timestamp="]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    # Last record in the file.
+    assert re.search(r"lsn: 12", output)
+
+
+@pytest.mark.parametrize("input", ["1731592956.8182", "2024-11-14T14:02:36.818299999Z"])
+def test_cat_xlog_file_timestamp_float(tt_cmd, tmp_path, input):
+    # Copy the .xlog file to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "timestamp.xlog")
+    shutil.copy(test_app_path, tmp_path)
+
+    cmd = [tt_cmd, "cat", "timestamp.xlog", "--timestamp={0}".format(input)]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    assert re.search(r"lsn: 6", output)
+    assert re.search(r"timestamp: 1731592956.8181", output)
+    assert not re.search(r"lsn: 8", output)
+    assert not re.search(r"timestamp: 1731592956.8184", output)
+
+
+def test_cat_xlog_file_timestamp_rfc3339(tt_cmd, tmp_path):
+    # Copy the .xlog file to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_file", "timestamp.xlog")
+    shutil.copy(test_app_path, tmp_path)
+    cmd = [tt_cmd, "cat", "timestamp.xlog", "--timestamp=2024-11-14T14:02:35Z"]
+    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
+    assert rc == 0
+    assert not re.search(r"lsn: 6", output)
+    assert not re.search(r"timestamp: 1731592956.8181", output)
+    assert not re.search(r"lsn: 8", output)
+    assert not re.search(r"timestamp: 1731592956.8184", output)
```

Comment:
These tests can be merged into one using one more parameter case.

Reply:
All tests were merged.